app/javascript/components/shared_components/search/SearchBar.jsx

```diff
@@ -1,20 +1,39 @@
-import React from 'react';
+import React, { useEffect, useMemo } from 'react';
 import PropTypes from 'prop-types';
+import debounce from 'lodash.debounce';
+import { useTranslation } from 'react-i18next';
+
+export default function SearchBar({ searchInput, setSearchInput }) {
+  const { t } = useTranslation();
+
+  const onChangeHandler = (event) => {
+    setSearchInput(event.target.value);
+  };
+
+  const debouncedOnChangeHandler = useMemo(
```

Comment:
Isn't a `useCallback` more appropriate here if we'd like to memoize the handler (which may not be required)?
The event can be passed to the callback,

Reply:
useCallback would be used on every refetch

Reply:
Umm, what do you mean?

Reply:
I meant re-render, not refetch my bad.
https://dmitripavlutin.com/react-throttle-debounce/

Reply:
Umm, I still don't see how `each time the component re-renders, a new instance of the debounced function is created by the debounce(changeHandler, 300)` occurring.

Reply:
They're the same internally,
One will simply memoize the callback while the other memoizes its returned value,
FYI check `mountMemo, updateMemo, mountCallback and updateCallback` in https://github.dev/facebook/react/tree/main/packages/react/src